### ISSUE TITLE
iOS push not working after some time

### ISSUE THREAD

**Issue by ecky-l** [ChatSecure/ChatSecure-iOS#575]
When I open Chatsecure on the iPhone, it logs in to my account (my own xmpp server running Prosody). Then I close the app, i.e. put into background, not force close, and for some time it seems to stay online. During this time I get push notifications when trying to chat from another test account.

But after some time, ChatSecure disconnects my account. Then the push service doesn't work anymore.

I have Prosody 0.9.10 on CentOS 7 running and the following modules enabled in a virtual host:

modules_enabled = {
                "cloud_notify";
                "smacks";
                "smacks_offline";
                "pinger";
                "...some other..."
        }

When trying to chat to my ChatSecure account while it is offline, I get the following log messages in my prosody.log :

Oct 20 10:08:47 s2sout2c5d7e0   info    outgoing s2s stream [my server vip]->pubsub.chatsecure.org closed: stream closed
Oct 20 10:08:47 s2sout2c5d7e0   info    Sending error replies for 1 queued stanzas because of failed outgoing connection to pubsub.chatsecure.org
Oct 20 10:08:51 s2sout2aef5f0   info    Beginning new connection attempt to pubsub.chatsecure.org ([107.170.218.87]:5269)
Oct 20 10:08:53 [my server vip]:saslauth     info    SASL EXTERNAL with pubsub.chatsecure.org failed

Its an iPhone SE with iOS 10, if that matters. It would be nice to get push notifications even when ChatSecure has logged me out.

**Comment by chrisballinger** [comment]
This seems to be a common issue. Verify that your server's TLS certificate is valid (not self signed) and includes both client and server bits.

https://github.com/ChatSecure/ChatSecure-iOS/issues/557#issuecomment-247479188

**Comment by ecky-l** [comment]
The server cert was valid (startssl class 1), but indeed, the Client extension was missing. 
I generated a new certificate request and had it signed by StartSSL, now the client section is present when I execute

`openssl x509 -in yourcertificate.crt -text -noout`

X509v3 Extended Key Usage:
TLS Web Client Authentication, TLS Web Server Authentication

And I made it available to prosody. Restarted the server, logged out and in in ChatSecure (got a prompt to verify the new certificate), waited a while until automatic logoff... Then tried to chat from a test account. No push notification and in the logs still the same error:

`Oct 20 17:46:30 [my server vip]:saslauth	info	SASL EXTERNAL with pubsub.chatsecure.org failed`

Could it be that the new cert takes a while until it is valid? Or are Class1 certs from startssl not good enough? Otherwise I am running out of options...
